flexmeasures/cli/data_add.py

```diff
@@ -1157,3 +1220,52 @@ def parse_source(source):
     else:
         _source = get_or_create_source(source, source_type="CLI script")
     return _source
+
+
+def get_or_create_model(
+    model_class: Type[GenericAsset | GenericAssetType | Sensor], **kwargs
+) -> GenericAsset | GenericAssetType | Sensor:
+    """Get a model from the database or add it if it's missing.
+
+    For example:
+    >>> weather_station_type = get_or_create_model(
+    >>>     GenericAssetType,
+    >>>     name="weather station",
+    >>>     description="A weather station with various sensors.",
+    >>> )
+    """
+
+    # unpack custom initialization parameters that map to multiple database columns
+    init_kwargs = kwargs.copy()
+    lookup_kwargs = kwargs.copy()
+    if "knowledge_horizon" in kwargs:
+        (
+            lookup_kwargs["knowledge_horizon_fnc"],
+            lookup_kwargs["knowledge_horizon_par"],
+        ) = lookup_kwargs.pop("knowledge_horizon")
+
+    # Find out which attributes are dictionaries mapped to JSON database columns,
+    # or callables mapped to string database columns (by their name)
+    filter_json_kwargs = {}
+    filter_by_kwargs = lookup_kwargs.copy()
+    for kw, arg in lookup_kwargs.items():
+        model_attribute = getattr(model_class, kw)
+        if hasattr(model_attribute, "type") and isinstance(model_attribute.type, JSON):
+            filter_json_kwargs[kw] = filter_by_kwargs.pop(kw)
+        elif callable(arg) and isinstance(model_attribute.type, String):
+            filter_by_kwargs[kw] = filter_by_kwargs[kw].__name__
+
```

Comment:
Should there not be an `else` clause, possibly with an Exception thrown?
The code now silently ignores arguments here if they are not JSON or callable. Furthermore, it is still potentially using them in creation later on, which is a potential mismatch.
Finally, I don't get what the callable distinction is for. Please provide an example what you're going for here.